### DISCUSSION TITLE
Radios

### DISCUSSION THREAD

**Discussion by jrbarnes9** [ONSdigital/design-system#56] [Feedback]
Use this place to discuss the [radios component](https://ons-design-system.netlify.com/components/radios) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by will-madebymoon** [comment]
Something I've noticed which we've updated in our prototypes however hasn't been completed within the pattern library is the indentation of the description within the radio button.

Currently, the description is indented see screengrab. 
<img width="395" alt="screenshot 2019-01-22 at 15 34 26" src="https://user-images.githubusercontent.com/8184622/51546113-5d702a80-1e5b-11e9-9ea7-d316ece5c4e1.png">

The intention is to remove the indentation see screengrab.
<img width="685" alt="screenshot 2019-01-22 at 15 22 55" src="https://user-images.githubusercontent.com/8184622/51546145-711b9100-1e5b-11e9-9650-f6a3eb98f6a8.png">

This would also needed to be carried across checkboxes

**Reply by jrbarnes9** [reply]
#135 resolved

**Comment by will-madebymoon** [comment]
When radios or checkboxes are greater than the standard length the right padding should be 1rem from the label.

**Reply by jrbarnes9** [reply]
#207 resolved

**Comment by jrbarnes9** [comment]
Screen reader doesn't read out revealed text input field label or content when selecting 'other' radio option.

**Reply by jrbarnes9** [reply]
#272 resolved

**Comment by bameyrick** [comment]
We need to add a parameter to allow for an "Or" to placed between the main choices and an "other" choice.

![Screenshot 2019-04-01 at 10 15 15](https://user-images.githubusercontent.com/8447446/55316811-160fa700-5467-11e9-9d65-a77b1231b8e2.png)

**Reply by jrbarnes9** [reply]
#296 resolved

**Comment by bameyrick** [comment]
The attributes parameter currently has no affect.

**Reply by jrbarnes9** [reply]
#352 resolved

**Comment by bameyrick** [comment]
It would be good to also be able to set the checked radio from a top level `value` parameter. E.g.: https://nunjucks-macros-example.herokuapp.com/radio

**Reply by jrbarnes9** [reply]
#418 resolved

**Comment by bameyrick** [comment]
Radios break the field layout because the `field__items` element is a `div` rather than a `span`

**Reply by jrbarnes9** [reply]
#444 resolved

**Comment by bameyrick** [comment]
Theres no option to prevent radios from wrapping themselves in a fieldset.

**Reply by jrbarnes9** [reply]
#523 resolved

**Comment by trevorsaint** [comment]
I would expect the full area of the radio button to be clickable.
![image](https://user-images.githubusercontent.com/4989027/88305945-06723c80-cd02-11ea-8405-2c97ec5a8788.png)
Anything below the highlighted purple marking has no effect on selection.

A user may attempt to click / tap in that area.

**Reply by jrbarnes9** [reply]
#850 resolved

**Comment by jrbarnes9** [comment]
Update select/write-in examples on checkbox/radios

- Update 'select' examples so it shows best use case e.g. the select options are a sub-category group of a radio category (rather than just a list of 'other' categories)
- Hide all write-ins/selects in examples unless example is specifically showing the 'visible' variant.

**Reply by jrbarnes9** [reply]
#1077 resolved

**Comment by jrbarnes9** [comment]
Space between radio/checkbox label and description is not maintained when description wraps

![Screenshot 2020-11-20 at 12 06 23](https://user-images.githubusercontent.com/43346934/99798475-03108580-2b29-11eb-8d3d-2e38eb6b0826.png)

**Reply by jrbarnes9** [reply]
#1180 resolved

**Comment by jrbarnes9** [comment]
Remove comma after 'Or' from mutually exclusive radios

**Reply by jrbarnes9** [reply]
#1192 resolved

**Comment by jrbarnes9** [comment]
`fieldset__legend` markup and margin still shows even when not used in macro

**Reply by jrbarnes9** [reply]
#1263  resolved

**Comment by jrbarnes9** [comment]
Add error validation border to select when no option selected

Currently:
![Screenshot 2021-06-18 at 09 19 08](https://user-images.githubusercontent.com/43346934/122530652-9efa6500-d016-11eb-9cc8-69cb6a1ba1f8.png)

Should be:
![Screenshot 2021-06-18 at 09 19 58](https://user-images.githubusercontent.com/43346934/122530675-a588dc80-d016-11eb-9890-4220714cf348.png)

**Reply by jrbarnes9** [reply]
#1562 resolved

**Comment by jrbarnes9** [comment]
Convert radio and checkbox input sizes from rem to px to avoid decimal sizes

Ensure the selected state svg sits centred

**Reply by jrbarnes9** [reply]
#1562 resolved

**Comment by mr-nick17** [comment]
New variant for radios without a border, as per [DP's design system](https://dp-design-system.netlify.app/components/radios/). This variant is to be used on census web pages and other new pages that have a radio component.

![Screenshot 2021-08-05 at 10 09 55](https://user-images.githubusercontent.com/19624419/128830359-7d23a47b-7d1e-476b-9656-6a8060c2a567.png)

**Reply by jrbarnes9** [reply]
#1628 resolved

**Comment by jrbarnes9** [comment]
I think we should revert back to indenting the radio/checkbox descriptions so they align left with the labels.  When labels wrap, they create an unsightly alignment with the description
From:
![Screenshot 2021-08-13 at 09 02 49](https://user-images.githubusercontent.com/43346934/129325244-92f2a392-e4bd-442f-a6f2-8e1063c82b0e.png)
To:
<img width="365" alt="Screenshot 2021-09-14 at 09 32 01" src="https://user-images.githubusercontent.com/43346934/133224037-5deb8286-e83e-45f9-b4f0-88a180e91a64.png">

**Reply by jrbarnes9** [reply]
#1708 raised

**Comment by jrbarnes9** [comment]
Add checkboxes and radios as input types for the 'other' conditional reveal variant for radios and checkboxes components

<img width="705" alt="Screenshot 2021-08-18 at 15 54 35" src="https://user-images.githubusercontent.com/43346934/129921139-ecb8300d-69de-44f9-9221-918ca1aee251.png">

**Reply by jrbarnes9** [reply]
Issue #1648 raised

**Comment by mr-nick17** [comment]
The radio button component when selected and zoomed to less than 100% is not aligned correctly (see image)

![Screenshot 2023-03-16 at 10 36 09](https://user-images.githubusercontent.com/19624419/225591807-4a1828d0-5ad5-4efd-a455-8879f45cd696.png)

**Reply by mr-nick17** [reply]
Seems to be more of an issue on MacOS

**Reply by daniellecorke** [reply]
Hi Nick, thanks for highlighting this. I've raised #3133 to investigate.

**Reply by daniellecorke** [reply]
Hi Nick, this bug was fixed as part of [Release 70.0.4](https://github.com/ONSdigital/design-system/releases/tag/70.0.4). Please let us know if you have any further issues.

**Comment by katie-gardner** [comment]
Please could support be added for having textarea as the input type in the reveal variant for radios? One of the IDS screens is looking to have something like this
<img width="439" alt="radio_reveal_textarea" src="https://github.com/ONSdigital/design-system/assets/114991656/8b237439-30a1-4803-ad23-c8a0a73c910b">
Additionally, please could there be an option to have all the radios the same width, all matching the length of the longest? The IDS designs have multiple radio pages with all the fields aligned

**Reply by daniellecorke** [reply]
Hi Katie, thanks for your suggestions and feedback. I've added #3128 and #3129 to our backlog so that we can look into this for you.

**Reply by katie-gardner** [reply]
Great, thank you!

**Reply by daniellecorke** [reply]
Hi Katie, I'm pleased to say these changes have been released in [Release 70.0.4](https://github.com/ONSdigital/design-system/releases/tag/70.0.4). Please let us know if you have any further feedback.

**Reply by katie-gardner** [reply]
Will do - thank you!

**Comment by cconway-ext-ons** [comment]
Could I please request that a pre-set value be available in a texarea within a radio, which was added in Release 70.0.4?

The ids hub team would like to use this for a multiple page form, so that if you return to a previous page your inputs will be loaded from the session and displayed. 


The change I would like made is to add the line `"value": radio.other.value` to as line 133 in the file src/components/radios/_macro.njk , so that the text area elif text area part is

```
{% elif otherType == "textarea" %}
    {% from "components/textarea/_macro.njk" import onsTextarea %}
    {{
        onsTextarea({
            "id": radio.other.id,
            "name": radio.other.name,
            "label": {
                "id": radio.other.id + "-label",
                "text": radio.other.label.text,
                "classes": 'ons-u-fw-n',
                "description": radio.other.label.description
            },
            "charCheckLimit": {
                "limit": radio.other.charCheckLimit.limit,
                "charCountSingular": radio.other.charCheckLimit.charCountSingular,
                "charCountPlural": radio.other.charCheckLimit.charCountPlural
            },
            "value": radio.other.value
        })
    }}
{% endif %}
```

**Reply by MagdalenaLarge** [reply]
Hi, I would like to give you the update about your request.
It has been addressed to our dev team and fix has been included in our latest release: [70.0.9](https://github.com/ONSdigital/design-system/releases/tag/70.0.9).
Please let us know in case of any further questions.
Thank you.

**Comment by AlanHONS** [comment]
See https://onsdigital.slack.com/archives/C038E2C4K2R/p1730308652383799 for an Interaction Design discussion about Service Manual information text about when to use Boxed vs UnBoxed radio options. 
Highlighting here for Design System team to consider if they want to update that Service Manual wording for greater clarity?

**Reply by MagdalenaLarge** [reply]
Thank you @AlanHONS for your suggestion. We will look into it.

**Comment by matthew-robinson-ons** [comment]
Could i request the addition of a disabled option for radio components please. We have had a requirement to disable radio buttons for users without edit permissions, but still allow them to view the information the buttons display, the same way checkboxes work when disabled. This is currently not possible using design system radios.

**Reply by MagdalenaLarge** [reply]
Hello @matthew-robinson-ons ,
Thank you for contacting us. 
Are you able to provide more details about your requirement? Any visuals, how do you plan to use it? Was this solution tested with the users?
Thank you.

**Reply by matthew-robinson-ons** [reply]
Hello @MagdalenaLarge 

We would be looking for something like:

`{% if radio.disabled %}disabled{% endif %}`

So the user can still see if a radio is checked and the radio content, without being able to update it. 

Disabled:
<img width="352" height="232" alt="Screenshot 2025-08-11 at 10 12 13" src="https://github.com/user-attachments/assets/a15b1bd8-fe4c-4bab-834e-4d809805a58d" />

Enabled:
<img width="355" height="229" alt="Screenshot 2025-08-11 at 10 16 31" src="https://github.com/user-attachments/assets/41a67375-1f8b-43b9-954a-dbebd52630b8" />

This hasn't currently been solution tested with the users.

**Reply by MagdalenaLarge** [reply]
Hi @matthew-robinson-ons,
Thank you for the screenshots. I will have to review them with our Interaction Designer to ensure they meet our design best practices. There also concern regarding accessibility of this solution - potentially some screen readers would not be able to distinguish between enabled and disabled content of the radios. 
I was wondering if you considered using different component as checkbox?
May I also ask which product you are supporting in case of further questions. 
Thank you.

**Reply by matthew-robinson-ons** [reply]
Okay, thanks for the quick response. 
I'm working on RASRM. 
I would have to check as to whether we would be able to update this solution to use checkboxes but it was my understanding that checkboxes are preferred for multiple selections, this feature is designed for only one option to be selected by the user.

**Reply by MagdalenaLarge** [reply]
Thank you for the information. I will come back to you as soon as possible.

**Reply by MagdalenaLarge** [reply]
Hi @matthew-robinson-ons ,
I had a quick chat with out Interaction Designer regarding your request and we are still concerned that the proposed solution is not accessible and may be confusing for the customer. We would be happy to discuss details and support you with the option which would cover user needs (which we are still not sure about) and include the best design standards. I will send a quick invitation for September, so we can discuss more details. Thanks.
